tests/virtualization/universal/waitfor_guests.pm

```diff
@@ -102,15 +102,13 @@ sub run {
     add_guest_to_hosts $_, $virt_autotest::common::guests{$_}->{ip} foreach (keys %virt_autotest::common::guests);
     assert_script_run "cat /etc/hosts";
 
-    # Wait for guests to finish installation
-    # script_run("wait", timeout => 1200);
-    # script_retry("! ps x | grep -v 'grep' | grep 'virt-install'", retry => 120, delay => 10);
-    # Unfortunately this doesn't cover the second step of the installation, where ssh is available
-    my $sleep_delay = 1200;
-    $sleep_delay = 2400 if (is_xen_host);    # XEN has more guests
-    sleep($sleep_delay);    # XXX Get rid of this sleep!
-    wait_guest_online($_, 120) foreach (keys %virt_autotest::common::guests);
-    record_info("guests installed", "Guest installation completed");
+    # Wait for guests to announce that installation is complete
+    script_retry("cat guests_log", retry => 40, delay => 60);
```

Comment:
Is this printing the guests_log initial content to screen?

Reply:
Yes, it will print it.

1. Fully replacing ensure_online wasn't my intention in this patch, but I think is possible.
2. We wait for the first successful guest install + extra 10 minutes for other guests. For example third guest [failed here](http://d488.qam.suse.de/tests/1605#step/waitfor_guests/128) and whole module took 24 minutes on our old server. I think it is nice improvement.
3. Do you advice to create different keys for all the guests(sle12sp3,sle12sp4... etc) or just not use the same key on host and guests?

I think we can continue in progress if we want to replace fully the ensure_online step with the after.local solution.